lib/controllers/root-controller.js

```diff
@@ -574,6 +587,44 @@ export default class RootController extends React.Component {
     viewElement.focus();
   }
 
+  viewChangesForCurrentFile(stagingStatus) {
+    const editor = this.props.workspace.getActiveTextEditor();
+    const absFilePath = editor.getPath();
+    const repoPath = this.props.repository.getWorkingDirectoryPath();
+    if (absFilePath.startsWith(repoPath)) {
+      const filePath = absFilePath.slice(repoPath.length + 1);
+      this.quietlySelectItem(filePath, stagingStatus);
+      const splitDirection = this.props.config.get('github.viewChangesForCurrentFileDiffPaneSplitDirection');
+      const pane = atom.workspace.getActivePane();
+      if (splitDirection === 'right') {
+        pane.splitRight();
+      } else if (splitDirection === 'down') {
+        pane.splitDown();
+      }
+      return this.diveIntoFilePatchForPath(filePath, stagingStatus, {
+        lineNumber: editor.getCursorBufferPosition().row + 1,
+        amending: stagingStatus === 'staged' && this.state.isAmending,
+      });
+    } else {
+      throw new Error(`${absFilePath} does not belong to repo ${repoPath}`);
+    }
+  }
+
+  @autobind
+  viewUnstagedChangesForCurrentFile() {
+    this.viewChangesForCurrentFile('unstaged');
+  }
+
+  @autobind
+  viewStagedChangesForCurrentFile() {
+    this.viewChangesForCurrentFile('staged');
+  }
+
+  @autobind
+  goToDiffLine(lineNumber) {
+    return this.filePatchController.goToDiffLine(lineNumber);
+  }
```

Comment:
Yeah this felt icky to me. IIRC @BinaryMuse looked into Redux and dismissed using it for some reason. Can't remember what